plugin/src/main/java/org/opensearch/ml/action/mcpserver/McpAsyncServerHolder.java

```diff
@@ -22,15 +23,57 @@
  */
 public class McpAsyncServerHolder {
 
-    public static OpenSearchMcpServerTransportProvider mcpServerTransportProvider = new OpenSearchMcpServerTransportProvider(
-        new ObjectMapper()
-    );
-
     public static Map<String, StreamingRestChannel> CHANNELS = new ConcurrentHashMap<>();
+    public static Map<String, Long> IN_MEMORY_MCP_TOOLS = new ConcurrentHashMap<>();
+    private static volatile MLIndicesHandler mlIndicesHandler;
+    private static volatile McpToolsHelper mcpToolsHelper;
+
+    public static void init(MLIndicesHandler mlIndicesHandler, McpToolsHelper mcpToolsHelper) {
```

Comment:
nitialization Order** - No clear contract about when init() must be called. Can we please put all the variables first?

Reply:
Also do we need to check null for these parameters?